src/test/java/ibm/jceplus/junit/base/BaseTestXDH.java

```diff
@@ -336,20 +337,10 @@ private void runSmallOrderTest() throws Exception {
     private void testSmallOrder(String name, String a_pri, String b_pub, String result)
             throws Exception {
 
-        try {
-            //System.out.println("Pub - "+b_pub);
+        Exception thrown = Assertions.assertThrows(IllegalStateException.class, () -> {
```

Comment:
What about the other types of exceptions that were caught before? Are they not a possibility anymore?

Reply:
With the updates in place when providing a small value as this test is intending we should only get an `IllegalStateException` now with the given updates in the C method which now returns the exception. Also please note i needed to remove the Junit 5 syntax in this spot since it was failing with an error `IllegalAccess Class 'class ibm.jceplus.junit.base.BaseTestXDH' no access to: class 'interface org.junit.jupiter.api.function.Executable'` which I have not been able to resolve. Syntax was reverted back to try catch as was present previously.